plugin-self-test

```diff
@@ -1,16 +1,20 @@
 #!/bin/sh
 
-# Build the dependency extractor plugin
-./gradlew :plugin:build
+# Build and publish the dependency graph plugin
+./gradlew :plugin:publishPluginMavenPublicationToLocalPluginsRepository
 
 # Invoke a build extracting the GitHub Dependency Graph
 ./gradlew --no-configuration-cache --no-build-cache -I plugin-self-test.init.gradle --console=plain $*
 
-# Submit the dependency graph to GitHub
-curl -L \
-  -X POST \
-  -H "Accept: application/vnd.github+json" \
-  -H "Authorization: Bearer $GITHUB_TOKEN"\
-  -H "X-GitHub-Api-Version: 2022-11-28" \
-  https://api.github.com/repos/gradle/github-dependency-extractor/dependency-graph/snapshots \
-  -d @build/reports/github-dependency-report/github-dependency-manifest.json
+if [[ -z "${GITHUB_TOKEN}" ]]; then
+  echo "GITHUB_TOKEN must be set in order to submit dependency graph"
+else
+  # Submit the dependency graph to GitHub
+  curl -L \
+    -X POST \
+    -H "Accept: application/vnd.github+json" \
+    -H "Authorization: Bearer $GITHUB_TOKEN"\
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    https://api.github.com/repos/gradle/github-dependency-extractor/dependency-graph/snapshots \
+    -d @build/reports/github-dependency-report/github-dependency-manifest.json
```

Comment:
Yes, to start with the submission will be done directly in the GitHub Action, since this is trivial to do.

It will be straightforward to later create a separate plugin to do the submission of a generated dependency graph file. I think it's important to separate the process of generating the file from the submission to GitHub, in order to reduce the work that is done in the `BuildOperationListener` at completion of the build.

Reply:
I removed the submission code quite some time ago, since it wasn't working with Gradle 8.0 and I wasn't able to quickly resolve the issue.